docs/README.md

```diff
@@ -0,0 +1,423 @@
+# Documentation for Setec
+
+## Table of Contents
+
+- [Background](#background)
+- [Basic usage](#basic-usage)
+- [Migrating to Setec](#migrating-to-setec)
+- [Operations and Maintenance](#operations-and-maintenance)
+   - [Secret Rotation](#secret-rotation)
+   - [Automatic Updates](#automatic-updates)
+   - [Bootstrapping and Availability](#bootstrapping-and-availability)
+- [Testing](#testing)
+- [API documentation](api.md)
+
+## Background
+
+Programs in production often need access to passwords, API keys, parameterized
+connection URLs, and other sensitive information (hereafter, **secrets**) at
+runtime.  Securely deploying secrets in production complicates operations, and
+typically involves a combination of tedious manual intervention (e.g., copying
+secrets out of secure storage at startup), or integrating with complex secrets
+management infrastructure (usually specific to the deployment environment).
+
+Setec comprises a lightweight HTTP-based [API](api.md) and a corresponding
+server, that allows programs running on a tailnet to securely fetch secrets at
+runtime. Access to secrets is governed by the tailnet's policy document, and
+the server maintains secrets in encrypted storage, keeps an audit log of
+accesses, and manages periodic backups.
+
+The setec server integrates with existing key-management infrastructure to
+bootstrap its own deployment (as of 24-Sep-2023, the AWS KMS is supported).
+Once the server is running on a tailnet, other programs can use it to access
+their production secrets with a basic WireGuard-encrypted HTTP request, rather
+than having to distribute secrets via files, environment variables, or manual
+operator intervention.
+
+In addition to reducing deployment toil, this also helps reduce the attack
+surface for managing secrets in third-party deployment environments: All the
+secrets are stored in one place, with access controls and audit logs to allow
+forensics in the event of a compromise.
+
+## Basic usage
+
+All of the methods of the setec API are HTTPS POST requests. Calls to the API
+must include a `Sec-X-Tailscale-No-Browsers: setec` header.  This prevents
+browser scripts from initiating calls to the service. The examples below assume
+you have a setec server running at `secrets.example.ts.net` on your tailnet.
+
+Go programs can use the [setec client library][setecclient] provided in this
+repository. The API supports any language, however, so the examples below are
+written using the curl command-line tool.
+
+A program that wishes to fetch a secret at runtime does so by issuing an HTTPS
+POST request to the [`/api/get` method] of the secrets service.
+
+For example:
+
+```shell
+curl -H sec-x-tailscale-no-browsers:setec -H content-type:application/json -X POST \
+  https://secrets.example.ts.net/api/get -d '{"name": "prod/myprogram/secret-name"}'
```

Comment:
Ugh. Yeah, and I think we should actually add tags to those instead, because it's weird for everyone _not_ in Go, for which that isn't conventional (and not all JSON implementations are magically case-insensitive).

But for now I'll make them match the declaration, and fix that separately 🙂

Reply:
(done)